app/models/bookmark.rb

```diff
@@ -12,4 +12,47 @@ class Bookmark < ApplicationRecord
   def document_type
     SolrDocument
   end
+
+  def self.destroy_without_solr_documents
+    batch_size = Orangelight.config["bookmarks"]["batch_size"]
+    Bookmark.find_in_batches(batch_size:).with_index do |bookmarks, batch|
+      Rails.logger.info { "Processing destroy_without_solr_documents group ##{batch}" }
+      bookmark_doc_ids = bookmark_doc_ids(bookmarks)
+      doc_ids_without_solr_doc = bookmark_doc_ids - doc_ids_in_solr(bookmark_doc_ids)
+
+      if doc_ids_without_solr_doc.present?
+        doc_ids_without_solr_doc.each do |doc_id|
+          Bookmark.where(document_id: doc_id)&.destroy_all
+        end
+      end
+    end
+  end
+
+  def self.update_to_alma_ids
+    Bookmark.where("LENGTH(document_id) <= 7").find_in_batches.with_index do |bookmarks, batch|
```

Comment:
For peace of mind, could we also confirm that this won't attempt to convert SCSB-*, coin-*, and dsp* ids to alma ids?  I don't think we have one, but theoretically, a document with id:coin-1 would also appear in these results.

Reply:
Thanks, good catch, updated.